patternlab/docs/a11y-checklist/headings.md

```diff
@@ -0,0 +1,94 @@
+[⬅️ a11y Checklist](a11y-checklist.md)
+
+# Headings `<h#>`
+
+## Heading level
+
+A heading needs its **proper level number** to represent the hierarchy of the content in the relationship of the parent and child components.
+
+HTML5 allows you to start with `<h1>` in every `<section>`. That is valid as HTML5, but it doesn't help assistive technology users to grasp the content hierarchy in the page.
+
+### ➤➤ Is the heading level number flexible?
+
+☑️ It can receive and consume the level number value from the parent component.
+
+☑️ It can have proper level number depending on where the component is placed in a page in the sequential manner from small to large. (If its parent component has *h3*, this component starts with *h4*.)
+
+The *sequential manner* means:
+
+☑️ No skipping - Some in-between number(s) is(are) NOT missing.
+
+☑️ No jumping - NOT putting heading level numbers randomly.
+
+<!--
+- [ ] No backward - Numbers does NOT go down from large to small.
+-->
+
+**➣ Yes:** The heading level number is set dynamically, and meets these conditions.
+
+**🛑 Watch out when your template contains multiple headings.** The heading levels could change in the template. You might need to adjust the level in the template.
```

Comment:
Sure.  This sometimes happens in organism level templates.  The template has a heading for its own.  It's not necessarily hardcoded, it might calls a heading template in it.  After that the template calls some child component templates, which contain their headings. Thus, one template contains multiple headings.  

The child components' heading levels are different from one for the heading for the initial template. So, the heading level value adjustment needs to happen between the heading for the initial template and the child components' in the initial template, not in its child component templates.

**Example:  03-organisms/by-author/about-section.twig** (note: the current heading levels are not correct in Mayflower.)

At L. 4, it calls **@atoms/04-headings/comp-heading.twig** for a heading for the about section as **\<h2\>** for example.

At L. 22, it calls **@organisms/by-author/person-card.twig** which contains a heading for person card. Since this is a child component of the about section, its heading should be **\<h3\>** by increasing its level by 1.  
This adjustment of the _level value_ should happen in **about-section.twig** (= parent component), not in its child component **@organisms/by-author/person-card.twig**, then the value is passed to the child component because the adjustment is only necessary in this context of the use of the 2 templates as the parent-child relationship. 

If this child component is used independently somewhere else and it increases its heading level value by 1 in the template, then its heading level could skip when another parent template with a different context from above passes its heading level value.

**Other samples:**

- 03-organisms/by-author/collapsible-content.twig
- 03-organisms/by-author/contact-list.twig (this is a working sample.)